src/main/java/net/mcreator/minecraft/ElementUtil.java

```diff
@@ -114,7 +114,7 @@ public static List<MCItem> loadBlocks(Workspace workspace) {
 		List<MCItem> elements = new ArrayList<>();
 		workspace.getModElements().stream().filter(element -> element.getType().getBaseType() == BaseType.BLOCK)
 				.forEach(modElement -> elements.addAll(
-						modElement.getMCItems().stream().filter(e -> !e.getName().endsWith(".bucket")).toList()));
```

Comment:
Isn't this type intentionally bucket for GeneratorWrapper#isBlock and GeneratorWrapper#isRecipeTypeBlockOrBucket and GeneratorWrapper#getElementPlainName

Reply:
The selector would get all the MCItems from fluids, but could differentiate the bucket only by its name, not by its type. With the new constructor, this hack isn't needed anymore

Reply:
Shouldn’t those methods then either be corrected or removed if not in use anymore?

Reply:
Those are still used by the generator in `mcitems.ftl` to properly map blocks/items, since the only context it has is the MCItem's name. Though it should be possible to remove the recipe type check by getting all the custom MCItems, then checking if the one with the same name is of type "block"